src/DotNetWorker/Hosting/WorkerHostBuilderExtensions.cs

```diff
@@ -171,6 +171,10 @@ public static IHostBuilder ConfigureFunctionsWorkerDefaults(this IHostBuilder bu
 
                     // Add default middleware
                     appBuilder.UseDefaultWorkerMiddleware();
+                })
+               .UseDefaultServiceProvider((context, options) =>
+                {
+                    options.ValidateScopes = context.HostingEnvironment.IsDevelopment();
```

Comment:
This looks like the correct change to me. What is the factory approach you mentioned in the PR description? How does that code change will look like?

Reply:
I think the factory approach would look something like this:

```csharp
.UseServiceProviderFactory(new AutofacServiceProviderFactory())
    .ConfigureContainer<ContainerBuilder>(builder =>
    {
        new DefaultServiceProvider(yourOptionsHere);
    });
```

But that doesn't feel right to me, a little convoluted? And I don't know the perk of doing it this way either